services/requester/remote_cadence_arch.go

```diff
@@ -0,0 +1,136 @@
+package requester
+
+import (
+	"context"
+	_ "embed"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/onflow/cadence"
+	errs "github.com/onflow/flow-evm-gateway/models/errors"
+	evmImpl "github.com/onflow/flow-go/fvm/evm/impl"
+	evmTypes "github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+	gethCommon "github.com/onflow/go-ethereum/common"
+	"github.com/onflow/go-ethereum/core/types"
+	"github.com/onflow/go-ethereum/crypto"
+)
+
+var (
+	//go:embed cadence/dry_run.cdc
+	dryRunScript []byte
+)
+
+var cadenceArchAddress = gethCommon.HexToAddress("0x0000000000000000000000010000000000000001")
+
+type RemoteCadenceArch struct {
+	blockHeight uint64
+	client      *CrossSporkClient
+	chainID     flow.ChainID
+	cachedCalls map[string]evmTypes.Data
+}
+
+var _ evmTypes.PrecompiledContract = (*RemoteCadenceArch)(nil)
+
+func NewRemoteCadenceArch(
+	blockHeight uint64,
+	client *CrossSporkClient,
+	chainID flow.ChainID,
+) *RemoteCadenceArch {
+	return &RemoteCadenceArch{
+		blockHeight: blockHeight,
+		client:      client,
+		chainID:     chainID,
+		cachedCalls: map[string]evmTypes.Data{},
+	}
+}
+
+func (rca *RemoteCadenceArch) Address() evmTypes.Address {
+	return evmTypes.NewAddress(cadenceArchAddress)
+}
+
+func (rca *RemoteCadenceArch) RequiredGas(input []byte) uint64 {
+	evmResult, err := rca.runCall(input)
+	if err != nil {
+		return 0
+	}
+
+	return evmResult.GasConsumed
+}
+
+func (rca *RemoteCadenceArch) Run(input []byte) ([]byte, error) {
+	key := hex.EncodeToString(crypto.Keccak256(input))
+
+	if result, ok := rca.cachedCalls[key]; ok {
+		return result, nil
+	}
+
+	evmResult, err := rca.runCall(input)
+	if err != nil {
+		return nil, err
+	}
+	return evmResult.ReturnedData, nil
+}
+
+func (rca *RemoteCadenceArch) runCall(input []byte) (*evmTypes.ResultSummary, error) {
+	tx := types.NewTx(
+		&types.LegacyTx{
+			Nonce:    0,
+			To:       &cadenceArchAddress,
+			Value:    big.NewInt(0),
+			Gas:      155_000,
```

Comment:
can we add a named constant for this.

Reply:
Good point :+1: 
Added in https://github.com/onflow/flow-evm-gateway/pull/635/commits/35818c32d9866f994b81f2a5e098ea34d699263e